posthog/tasks/update_cache.py

```diff
@@ -49,14 +53,47 @@
     CacheType.PATHS: Paths,
 }
 
+IN_A_DAY = 86_400
+
+
+def active_teams() -> List[int]:
+    """
+    Teams are stored in a sorted set. [{team_id: score}, {team_id: score}].
+    Their "score" is the number of seconds since last event.
+    Lower is better.
+    This lets us exclude teams not in the set as they don't have recent events.
+    That is, if a team has not ingested events in the last seven days, why refresh its insights?
+    And could let us process the teams in order of how recently they ingested events.
+    This assumes that the list of active teams is small enough to reasonably load in one go.
+    """
+    redis = get_client()
+    all_teams: List[Tuple[bytes, float]] = redis.zrange(RECENTLY_ACCESSED_TEAMS_REDIS_KEY, 0, -1, withscores=True)
+    if not all_teams:
+        teams_by_recency = sync_execute(
+            """
+            SELECT team_id, date_diff('second', max(_timestamp), now()) AS age
+            FROM events
+            WHERE _timestamp > date_sub(DAY, 3, now())
+            GROUP BY team_id
+            ORDER BY age;
+        """
+        )
+        redis.zadd(RECENTLY_ACCESSED_TEAMS_REDIS_KEY, {team: score for team, score in teams_by_recency})
+        redis.expire(RECENTLY_ACCESSED_TEAMS_REDIS_KEY, IN_A_DAY)
```

Comment:
Does this mean that any teams who join with "IN_A_DAY" will be excluded from refreshes? 

Also what does the recency do for us?

Reply:
> Does this mean that any teams who join with "IN_A_DAY" will be excluded from refreshes?

Yep, it would... so we could check more frequently..?

> Also what does the recency do for us?

Right now the recency only lets us include or exclude. But thinking ahead two steps ;) to #10991 since we'd have all candidate teams in a sorted set anyway. It would let us include recency of events as a weighting

Reply:
Yep, if we request an insight and it isn't in the cache we return "empty results" to the UI which requests the insight data